src/http/Message.h

```diff
@@ -123,6 +123,7 @@ class Message : public RefCountable
     /**
      * Validate the message start line is syntactically correct.
      * Set HTTP error status according to problems found.
+     * This method must return false if it is not passed parsed headers.
```

Comment:
This is wrong.

This method is a legacy code check that the first-line of a message fits HTTP syntax. It is a preparation step before passing the message to legacy HTTP parse. As such if the headers are already parsed successfully by any mechanism this method is irrelevant.

Reply:
OK. Then, I do not know what this meant:
>The suggested assertion is essential here and also requires sanityCheckStartLine() API documentation changes in http/Message.h. Nothing in the current sanityCheckStartLine() description says that the method must treat unparsed headers as insane (i.e. return false

Reply:
```C++
/**
 * Validate the message start line is syntactically correct.
 * Set HTTP error status according to problems found.
 *
 * \retval true   Status line has no serious problems.
 * \retval false  Status line has a serious problem. Correct response is indicated by error.
 */
virtual bool sanityCheckStartLine(const char *buf, const size_t hdr_len, Http::StatusCode *error) = 0;
```

I think there is a slight misunderstanding here due to vague terminology and my sloppy use of that terminology ("parsed", "headers", "status line", hdr_len, etc.). To address my concern without triggering Amos' (correct) objection, the added comment may say something like

```
* Zero-size hdr_len is treated as a serious problem.
```

The remaining method description already documents what happens when the method detects a serious problem.